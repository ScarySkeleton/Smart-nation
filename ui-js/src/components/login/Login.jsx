import React from 'react';

const Login = () => {
    return (
        <div className='container login'>
            Login page\popup
        </div>
    )
}

export default Login;
