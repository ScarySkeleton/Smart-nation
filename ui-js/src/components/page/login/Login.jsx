import React from 'react';

const Login = (props) => {
    return (
        <div className='container login'>
            Login page\popup
        </div>
    )
}

export default Login;
