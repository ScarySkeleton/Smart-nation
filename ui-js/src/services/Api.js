const API_PATH = '/';

function statusChecker(response) {
    if(response.status >= 200 && response.status < 300)
        return Promise.resolve(response);
    return Promise.reject(response);
}

/*
    ================================================================
                    USER LOGIN\LOGOUT\REGESTRATION
    ================================================================
*/
export function loginRequest(userData) {
    return function () {
        return fetch("http://localhost:50363/Account/Login", {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include",
            body: JSON.stringify(userData)
        })
        .then(statusChecker)
        .then(response => response.json())
        .then(json => {
            return {
                username: json.login
                , name: json.name
                , surname: json.surname
                , role: json.role
        }});
    }
};

export function registrationRequest(userData) {
    return function() {
        return fetch("http://localhost:50363/Account/Register", {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include",
            body: JSON.stringify(userData)
        })
        .then(statusChecker);
    }
}

export function logoutRequest() {
    return function() {
        return fetch("http://localhost:50363/Account/Logout", {
            method: "POST", 
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include",
        });
    }
}

/*
    ================================================
                    USER CABINET
    ================================================
*/
export function getCabinetData(data) {
    return function() {
        return {
            userId: 1,
            photo: null,
            description: 'My name is user and I like to read',
        }
    }
}

export function addBook(data) {
    return function() {
        return fetch("http://localhost:50363/PersonalCabinet/AddBook", {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include",
            body: JSON.stringify(data)
        })
        .then(statusChecker)
        .then(response => response.json())
        .then(json => {
            console.log(json);
        })
    }
}

export function getBookShelfBooks() {
    return function() {
        return fetch("http://localhost:50363/PersonalCabinet/GetBook", {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include"
        })
        .then(statusChecker)
        .then(response => response.json())
        .then(json => {
            console.log(json);
            return json;
        })
    }
}
