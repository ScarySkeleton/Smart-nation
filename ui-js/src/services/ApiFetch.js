const API_PATH='http://localhost:50363';

function statusChecker(response) {
    if(response.status >= 200 && response.status < 300)
        return Promise.resolve(response);
    return Promise.reject(response);
}

export function apiAuthPostFetch(path, data) {
    return function() {
        return fetch(`${API_PATH}/${path}`, {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            credentials: "include",
            body: JSON.stringify(data)
        })
        .then(statusChecker)
        .then(response => response.json())
        .then(json => {
            return json;
        });
    }
}

export function apiNonAuthPostFetch(path, data) {
    return function() {
        return fetch(`${API_PATH}/${path}`, {
            method: "POST",
            headers: {
                Accept: 'application/json, text/javascript, */*; q=0.01',
                'Content-type': 'application/json; charset=UTF-8',
            },
            mode: "cors",
            body: JSON.stringify(data)
        })
        .then(statusChecker)
        .then(response => response.json())
        .then(json => {
            return json;
        });
    }
}