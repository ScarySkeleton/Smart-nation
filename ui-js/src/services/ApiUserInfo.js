import {Error} from '../components/popup/Error/Error';
import {popupOpen} from '../components/popup/popup.action';
import store from './store/index';
import React from 'react';
import {connect} from 'react-redux';

/*
    ================================================
                    USER INFO
    ================================================
*/
const errorTitleMessage = 'Error was occured.'
const errorBodyMessage = `Error was occured while
    fetching data from server.`;



let fetchingError = (error) => {
    const errorPopup = {
        title: errorTitleMessage,
        body: <Error message={`${errorBodyMessage}
        Error message: ${error}`} /> 
    }
    
    return store.dispatch(popupOpen(errorPopup));
}


export {fetchingError};
