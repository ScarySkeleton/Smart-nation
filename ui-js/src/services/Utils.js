import Adapter from 'enzyme-adapter-react-15';
import { configure } from 'enzyme';

import valid from './validate/validate';

export function enzymeConf() {
    return configure({ adapter: new Adapter() });
}

export function validate(data, config) {
    valid.config = config;
    return valid.validate(data);
}
