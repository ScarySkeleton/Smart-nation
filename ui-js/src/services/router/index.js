import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import throttle from 'lodash/throttle';

import rootReducer from './rootReducer';
import rootSaga from './rootSaga';

import { loadState, saveState } from '../localStorage/localStorage';

const savedState = loadState();
const sagaMiddleware = createSagaMiddleware();
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;    

const store = createStore(
    rootReducer
    , savedState
    , composeEnhancers(applyMiddleware(sagaMiddleware)));

store.subscribe(throttle(() => {
    saveState({
        Login: store.getState().Login,
    })
}, 1000));
sagaMiddleware.run(rootSaga);

export default store;
