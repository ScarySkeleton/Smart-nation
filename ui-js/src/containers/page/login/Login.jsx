import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { withRouter, Redirect } from 'react-router';
import { Link } from 'react-router-dom';

import {loginRequest} from './login.actions';
import { valid } from '../../../services/Utils';
import {Facebook} from './Facebook/Facebook';
import './login.scss';

class LoginContainer extends PureComponent {
     constructor(props) {
         super(props);

         this.state = {
             userLogInfo: '',
             userPassword: '',
             isLogined: this.props.isLogined,
         }

         this.userLogInfoChange = this.userLogInfoChange.bind(this);
         this.userPasswordChange = this.userPasswordChange.bind(this);
         this.loginRequest = this.loginRequest.bind(this);
     }

     userLogInfoChange(e) {
         this.setState({
             ...this.state,
             userLogInfo: e.target.value,
         })
     }

     userPasswordChange(e) {
         this.setState({
             ...this.state,
            userPassword: e.target.value,
         })
     }

     loginRequest() {

        const data = {
            userLogInfo: this.state.userLogInfo,
            Password: this.state.userPassword,
        }

        const config = {
            userLogInfo: "isNonEmpty",
            userPassword: "isNonEmpty",
        }

        let toValidate = valid(data, config);
        if(toValidate.validate()) {
            return;
        }

        this.props.loginRequest(data);
     }

    render() {
        
        if(this.props.isLogined) {
            return <Redirect to="/" />;
        }

        return (
            <div className='login__page'>
                <div className='container login'>

                    <div className='login__title'>
                        <h2>Login</h2>
                    </div>

                    <div className='login__form'> 
                        <div className='login__block'>
                            <input className='login__block-input'
                             type='text' name='login'
                             onChange={this.userLogInfoChange} placeholder='LOGIN/PHONE'/>
                        </div>
                        
                        <div className='login__block'>
                            <input className='login__block-input'
                             type='password' name='password'
                             onChange={this.userPasswordChange} placeholder='PASSWORD' />
                        </div>
                    </div>
                    <div className='login__buttons'>
                        <button className='login__block-btn' onClick={this.loginRequest}>
                            LOGIN
                        </button>
                        <Link to='/registration' className='login__button__link'> Don't have account </Link>
                    </div>
                </div>
                <div className='login__facebook'>
                    <Facebook />
                </div>
            </div>
        );
    };
}

const mapStateToProps = state => {
    return {
        isLogined: state.Login.isLogined,
    }
}

const mapDispatchToProps = dispatch => {
    return {
        loginRequest: (data) => {
            dispatch(loginRequest(data));
        },
    }
}

const Login = withRouter(connect(
    mapStateToProps,
    mapDispatchToProps,
)(LoginContainer));
export default Login;
