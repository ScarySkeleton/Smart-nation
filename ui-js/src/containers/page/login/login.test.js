import React from 'react';
import { enzymeConf } from '../../../services/Utils';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from './Login';

describe('Login component ', () => {
    enzymeConf();
    it('should render correctly', () => {
        const tree = shallow(<Login />);
        expect(toJson(tree)).toMatchSnapshot();
    });
});
