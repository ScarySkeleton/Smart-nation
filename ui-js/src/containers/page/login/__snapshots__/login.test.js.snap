// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component  should render correctly 1`] = `
<div
  className="container login"
>
  Login page
  <button
    onClick={[Function]}
  >
    Try log in
  </button>
</div>
`;
